extensions/sql-database-projects/src/models/projectEntry.ts

```diff
@@ -124,6 +124,37 @@ export class SqlProjectReferenceProjectEntry extends FileProjectEntry implements
 	}
 }
 
+export class NugetPackageReferenceProjectEntry extends FileProjectEntry implements IDatabaseReferenceProjectEntry {
+	databaseSqlCmdVariableValue?: string;
+	databaseSqlCmdVariableName?: string;
+	databaseVariableLiteralValue?: string;
+	serverSqlCmdVariableName?: string;
+	serverSqlCmdVariableValue?: string;
+	suppressMissingDependenciesErrors: boolean;
```

Comment:
At some point, we should probably either

1. create `abstract UserDatabaseReferenceProjectEntry` that contains all the sqlcmdvar stuff, or
2. get rid of `FileProjectEntry` and just use the type structure that's been defined over in `mssql.d.ts` to cut out the conversion.

1 would be pretty easy to do now.  Do you think 2 makes sense (in the future), or is keeping 1 sufficient?

Reply:
1 makes sense. I'll send out a separate PR to make that change. For 2, I'm more hesitant because then we'd need more if/elses for ADS and vscode since the types are defined in `mssql.d.ts` and `vscode-mssql.d.ts`

Reply:
Yeahhh, though something I want to raise with Karl is how separate those files actually need to be.  Is there a reason we don't have `mssql-shared.d.ts` that gets consumed by both `mssql.d.ts` and `vscode-mssql.d.ts`?

Reply:
it's probably just because there hasn't been a need for sharing things between them, and maybe also because they could be using different versions of STS. Besides sql projects, there aren't any other extensions I know of that have this problem of being in both ADS and vscode and using STS calls and interfaces defined in those files.